python/metatensor-learn/tests/module_map.py

```diff
@@ -0,0 +1,143 @@
+
+            with torch.no_grad():
+                ref_values = module(block.values)
+            out_block = out_tensor.block(key)
+            assert torch.allclose(ref_values, out_block.values)
+
+            for parameter, gradient in block.gradients():
+                with torch.no_grad():
+                    ref_gradient_values = module(gradient.values)
+                assert torch.allclose(
+                    ref_gradient_values, out_block.gradient(parameter).values
+                )
+
+    @pytest.mark.parametrize(
+        "tensor",
+        [
+            random_single_block_no_components_tensor_map(HAS_TORCH, False),
+        ],
+    )
+    def test_linear_module_init(self, tensor):
+        tensor_module = Linear(tensor, tensor)
+        with torch.no_grad():
+            out_tensor = tensor_module(tensor)
+
+        for i, item in enumerate(tensor.items()):
+            key, block = item
+            module = tensor_module[i]
+            assert (
+                tensor_module.get_module(key) is module
+            ), "modules should be initialized in the same order as keys"
+
+            with torch.no_grad():
+                ref_values = module(block.values)
+            out_block = out_tensor.block(key)
+            assert torch.allclose(ref_values, out_block.values)
+            assert block.properties == out_block.properties
+
+            for parameter, gradient in block.gradients():
+                with torch.no_grad():
+                    ref_gradient_values = module(gradient.values)
+                out_gradient = out_block.gradient(parameter)
+                assert torch.allclose(ref_gradient_values, out_gradient.values)
+                assert gradient.properties == out_gradient.properties
+
+    @pytest.mark.parametrize(
+        "tensor",
+        [
+            random_single_block_no_components_tensor_map(HAS_TORCH, False),
+        ],
+    )
+    def test_linear_module_from_module(self, tensor):
+        tensor_module = Linear.from_module(
+            tensor.keys, in_features=len(tensor[0].properties), out_features=5
+        )
+        with torch.no_grad():
+            out_tensor = tensor_module(tensor)
+
+        for i, item in enumerate(tensor.items()):
+            key, block = item
+            module = tensor_module[i]
+            assert (
+                tensor_module.get_module(key) is module
+            ), "modules should be initialized in the same order as keys"
+
+            with torch.no_grad():
+                ref_values = module(block.values)
+            out_block = out_tensor.block(key)
+            assert torch.allclose(ref_values, out_block.values)
+
+            for parameter, gradient in block.gradients():
+                with torch.no_grad():
+                    ref_gradient_values = module(gradient.values)
+                assert torch.allclose(
+                    ref_gradient_values, out_block.gradient(parameter).values
+                )
```

Comment:
could you add a test checking that both ModuleMap and Linear can be compiled with `torch.jit.script`?

Reply:
And another test with `out_properties=...`

Reply:
Added tests for `out_properties` and also for `from_weight` in Linear